iwyu.cc

```diff
@@ -2906,7 +2915,18 @@ class InstantiatedTemplateVisitor
     // implicit stuff, it should not suggest forward declarations.
   }
 
-  void ReportTypeUse(SourceLocation used_loc, const Type* type) override {
+  void ReportTypeUse(SourceLocation used_loc, const Type* type,
+                     UseKind use_kind) override {
+    if (use_kind == UseKind::RemoveRefs ||
```

Comment:
`type` can never be nullptr here, I guess? I think `Resugar` and `CanIgnoreType` are both null-friendly.

Reply:
... so you may want to avoid `type->` before you know it's non-null.

Reply:
Let `ReportTypeUse` be null-friendly.